iothub/service/tests/QueryTests.cs

```diff
@@ -292,5 +293,59 @@ public void QueryResultUserSuppliedContinuationTest()
             Assert.AreEqual("test", r.ElementAt(0).DeviceId);
             Assert.IsTrue(q.HasMoreResults);
         }
+
+        [TestMethod]
+        public void QueryResult_OverrideDefaultJsonSerializer_Ok()
+        {
+            // simulate json serialize/deserialize
+            var serverQueryResult = new QueryResult()
+            {
+                Type = QueryResultType.Twin,
+                Items = new List<Twin>()
+                {
+                    new Twin()
+                    {
+                        DeviceId = "test",
+                    }
+                },
+                ContinuationToken = "GYUVJDBJFKJ"
+            };
+
+            var settings = new JsonSerializerSettings { MaxDepth = 3 };
+            // serialize
+            var jsonQueryResult = JsonConvert.SerializeObject(serverQueryResult, settings);
+            Assert.AreEqual("{\"type\":\"twin\",\"items\":[{\"deviceId\":\"test\",\"etag\":null,\"version\":null,\"properties\":{\"desired\":{},\"reported\":{}}}],\"continuationToken\":\"GYUVJDBJFKJ\"}", jsonQueryResult);
```

Comment:
Use FluentAssertions instead.

```suggestion
            jsonQueryResult.Should().Be("{\"type\":\"twin\",\"items\":[{\"deviceId\":\"test\",\"etag\":null,\"version\":null,\"properties\":{\"desired\":{},\"reported\":{}}}],\"continuationToken\":\"GYUVJDBJFKJ\"}");
```